lib/ansible/modules/windows/win_chocolatey_feature.ps1

```diff
@@ -0,0 +1,75 @@
+#!powershell
+# This file is part of Ansible
+
+# Copyright (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.CommandUtil
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+$ErrorActionPreference = "Stop"
+
+$params = Parse-Args -arguments $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+$name = Get-AnsibleParam -obj $params -name "name" -type "str" -failifempty $true
+$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "enabled" -validateset "disabled", "enabled"
+
+$result = @{
+    changed = $false
+}
+
+Function Get-ChocolateyFeatures {
+    param($choco_app)
+
+    $res = Run-Command -command "`"$($choco_app.Path)`" feature list -r"
+    if ($res.rc -ne 0) {
+        Fail-Json -obj $result -message "Failed to list Chocolatey features: $($res.stderr)"
+    }
+    $feature_info = @{}
+    $res.stdout -split "`r`n" | Where-Object { $_ -ne "" } | ForEach-Object {
+        $feature_split = $_ -split "\|"
+        $feature_info."$($feature_split[0])" = $feature_split[1] -eq "Enabled"
+    }
+
+    return ,$feature_info
+}
+
+Function Set-ChocolateyFeature {
+    param(
+        $choco_app,
+        $name,
+        $enabled
+    )
+
+    if ($enabled) {
+        $state_string = "enable"
+    } else {
+        $state_string = "disable"
+    }
+    $res = Run-Command -command "`"$($choco_app.Path)`" feature $state_string --name `"$name`""
```

Comment:
The recommended command here would be:

`choco feature enable --name="'<name>'"`

Reply:
While the command you have will work, it is not best practice.

Reply:
I have a few questions

* Why the double quoting of the value, surely having separate arguments (that are correctly escaped) is a better option
* Is it a preference from Chocolatey from a docs and ensuring user's escape correctly or is the parser expecting this format

This is calling `CreateProcessW` directly so cmd shell quotes don't take effect here, only `"` is used in this layer when being parsed by Windows.

While in this module it isn't as important, other modules we have a special function that takes in an array of args and correctly escapes them so the program receive those args as written (without the user escaping them themselves). The Chocolatey preferred format `--key="'value'"` won't work well with the function we have while the `--key value` format is a lot easier.

Reply:
Due to the myriad of escaping, and quote removal, and potential for spaces in paths/names, etc. that happens when interacting between cmd and powershell, etc. our guidance/best practice is always to use the equivalent of this:

`choco feature enable --name="'<featureName>'"`

for each Chocolatey command.  We have found that this is the most resilient way to correctly pass the intended parameters into Chocolatey, and have them be recognised.

Arguably, this isn't needed here, since feature names will never have spaces, etc.  However, if someone were to look at this command, and how it is doing something, and then apply the same to another Chocolatey command, it might not work as intended.